File: ObjCSample/ObjCSample/SampleControllers/Native Ads/NativeAdCell.m
 //
//  NativeAdCell.m
//  ObjCSample
//
//  Created by Julien Gomez on 08/09/2015.
//  Copyright (c) 2018 Smart AdServer. All rights reserved.
//

#import "NativeAdCell.h"

@implementation NativeAdCell

- (void)awakeFromNib {
    [super awakeFromNib];
    [self setup];
}

- (void)setup {
    self.callToActionButton.titleLabel.adjustsFontSizeToFitWidth = YES;
}

- (void)setIconViewHidden:(BOOL)hidden {
    self.iconImageView.hidden = hidden;
    _titleLeadingSpaceConstraint.constant = hidden ? 2 : 52;
}

- (void)updateConstraintsWhenCoverAndMediaAreHidden:(BOOL)hidden {
    _iconTopSpaceConstraint.constant = hidden ? 12 : 2;
    _subtitleLeadingSpaceConstraint.constant = hidden ? _titleLeadingSpaceConstraint.constant : 2;
    _subtitleLabel.numberOfLines = hidden ? 1:3;
}

@end

File: SwiftSample/SwiftSample/SampleControllers/Native Ads/Cells/NativeAdCell.swift
//
//  SampleNativeAdView.swift
//  SwiftSample
//
//  Created by Thomas Geley on 21/07/2016.
//  Copyright © 2016 Smart AdServer. All rights reserved.
//

import Foundation
import SASDisplayKit

class NativeAdCell: UITableViewCell {
    
    @IBOutlet weak var mediaView: SASNativeAdMediaView?
    @IBOutlet weak var titleLabel: UILabel?
    @IBOutlet weak var subtitleLabel: UILabel?
    @IBOutlet weak var iconImageView: UIImageView?
    @IBOutlet weak var coverImageView: UIImageView?
    @IBOutlet weak var callToActionButton: UIButton?
    @IBOutlet weak var adChoicesButton: SASAdChoicesView!
    
    @IBOutlet weak var titleLeadingSpaceConstraint: NSLayoutConstraint!
    @IBOutlet weak var subtitleLeadingSpaceConstraint: NSLayoutConstraint!
    @IBOutlet weak var iconTopSpaceConstraint: NSLayoutConstraint!
    
    class func loadFromNibNamed(_ nibNamed: String, bundle : Bundle? = nil) -> NativeAdCell? {
        return UINib(nibName: nibNamed, bundle: bundle).instantiate(withOwner: nil, options: nil)[0] as? NativeAdCell
    }
    
    func setIconViewHidden(_ hidden: Bool) {
        iconImageView?.isHidden = hidden
        titleLeadingSpaceConstraint.constant = hidden ? 2 : 52
    }
    
    func updateConstraintsWhenCoverAndMediaAreHidden(_ hidden: Bool) {
        iconTopSpaceConstraint.constant = hidden ? 12 : 2
        subtitleLeadingSpaceConstraint.constant = hidden ? titleLeadingSpaceConstraint.constant : 2
    }

}
